openpype/modules/ftrack/ftrack_module.py

```diff
@@ -479,6 +482,50 @@ def cli(self, click_group):
         click_group.add_command(cli_main)
 
 
+def _check_ftrack_url(url):
+    import requests
+
+    try:
+        result = requests.get(url, allow_redirects=False)
+    except requests.exceptions.RequestException:
+        return False
+
+    if (result.status_code != 200 or "FTRACK_VERSION" not in result.headers):
+        return False
+    return True
+
+
+def check_ftrack_url(url, log_errors=True, logger=None):
+    """Checks if Ftrack server is responding"""
+
+    if logger is None:
+        logger = Logger.get_logger(__name__)
+
+    url = url.strip("/ ")
+    if not url:
+        logger.error("Ftrack URL is not set!")
+        return None
+
+    if not url.startswith("http"):
+        url = "https://" + url
+
+    ftrack_url = None
+    if not url.endswith("ftrackapp.com"):
+        ftrackapp_url = url + ".ftrackapp.com"
+        if _check_ftrack_url(ftrackapp_url):
+            ftrack_url = ftrackapp_url
```

Comment:
I wonder why we shouldn't just enforce the user to put in full the url instead. So if they use a `ftrackapp.com` deployment why not have that included in the URL? Seems a bit magical almost?

I know it added it before already so it would somewhat break backwards compatibility? But maybe we should at least deprecate the automatic resolving?

Reply:
Not an issue or task for this PR.